Modules/ScormAicc/classes/SCORM/class.ilSCORMExplorer.php

```diff
@@ -298,135 +260,77 @@ function isClickable($a_type, $a_id = 0, $a_obj = 0)
-		$sc_object =& new ilSCORMItem($a_node_id);
-		$id_ref = $sc_object->getIdentifierRef();
-		$ilBench->stop("SCORMExplorer", "initSCORMItem");
-		
-		$ilBench->start("SCORMExplorer", "initResource");
-		//$sc_res =& new ilSCORMResource();
-		$sc_res_id = ilSCORMResource::_lookupIdByIdRef($id_ref, $sc_object->getSLMId());
-		$ilBench->stop("SCORMExplorer", "initResource");
-		
-		$scormtype = strtolower(ilSCORMResource::_lookupScormType($sc_res_id));
-		
-		$ilBench->start("SCORMExplorer", "renderLink");
-		$ilBench->start("SCORMExplorer", "renderLink_OutputIcons");
-		if ($this->output_icons)
-		{
-			if ($this->isClickable($a_option["c_type"], $a_node_id, $sc_object))
-			{
-				$this->getOutputIcons($tpl, $a_option, $a_node_id, $scormtype);
-			}
+		if ($this->output_icons && $clickable) {
+			$this->getOutputIcons($tpl, $option, $option["id"], $scormtype);
 		}
-		$ilBench->stop("SCORMExplorer", "renderLink_OutputIcons");
-		
-		if ($this->isClickable($a_option["c_type"], $a_node_id, $sc_object))	// output link
+
+		if ($clickable)	// output link
 		{
-			$ilBench->start("SCORMExplorer", "renderLink_parseLinkBlock");
 			$tpl->setCurrentBlock("link");
-			$frame_target = $this->buildFrameTarget($a_option["c_type"], $a_node_id, $a_option["obj_id"]);
+			$frame_target = $this->buildFrameTarget($option["c_type"], $option["id"], $option["obj_id"]);
 			if ($frame_target != "")
 			{
-				$tpl->setVariable("TITLE", ilUtil::shortenText($a_option["title"], $this->textwidth, true));
+				$tpl->setVariable("TITLE", ilUtil::shortenText($option["title"], $this->textwidth, true));
 				$tpl->setVariable("LINK_TARGET", "javascript:void(0);");
 				if ($jsApi == true) {
-					$tpl->setVariable("ONCLICK", " onclick=\"parent.API.IliasLaunch('".$a_node_id."');return false;\"");
+					$tpl->setVariable("ONCLICK", " onclick=\"parent.API.IliasLaunch('".$option["id"]."');return false;\"");
 				} else {
 					$tpl->setVariable("ONCLICK", " onclick=\"parent.APIFRAME.setupApi();parent.APIFRAME.API."
 						.($scormtype == 'asset' ? 'IliasLaunchAsset' : 'IliasLaunchSahs')
-						."('".$a_node_id."');return false;\"");
+						."('".$option["id"]."');return false;\"");
 				}
 			}
 			$tpl->parseCurrentBlock();
-			$ilBench->stop("SCORMExplorer", "renderLink_parseLinkBlock");
 		}
 		else			// output text only
 		{
 			$tpl->setCurrentBlock("text");
-			$tpl->setVariable("OBJ_TITLE", ilUtil::shortenText($a_option["title"], $this->textwidth, true));
+			$tpl->setVariable("OBJ_TITLE", ilUtil::shortenText($option["title"], $this->textwidth, true));
 			$tpl->parseCurrentBlock();
 		}
-		$ilBench->stop("SCORMExplorer", "renderLink");
-		
-		$ilBench->start("SCORMExplorer", "formatItemTable");
-		$this->formatItemTable($tpl, $a_node_id, $a_option["c_type"]);
-		$ilBench->stop("SCORMExplorer", "formatItemTable");
 
-		$tpl->setCurrentBlock("row");
+		$tpl->setCurrentBlock("li");
 		$tpl->parseCurrentBlock();
 
-		$this->output[] = $tpl->get();
+		return $tpl;
```

Comment:
Stimmt. Vorher wurde hier ein array mit vielen templates gefüllt. Jetzt haben wir nur noch ein template.
Das passt besser in ein ordentliches rekursive verhalten, was in der vorherigen version nicht der fall war.